htlcswitch/link.go

```diff
@@ -1920,13 +1920,26 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 		// As we've just accepted a new state, we'll now
 		// immediately send the remote peer a revocation for our prior
 		// state.
-		nextRevocation, currentHtlcs, err := l.channel.RevokeCurrentCommitment()
+		nextRevocation, currentHtlcs, finalHTLCs, err :=
+			l.channel.RevokeCurrentCommitment()
 		if err != nil {
 			l.log.Errorf("unable to revoke commitment: %v", err)
 			return
 		}
 		l.cfg.Peer.SendMessage(false, nextRevocation)
 
+		// Notify the incoming htlcs of which the resolutions were
+		// locked in.
+		for id, settled := range finalHTLCs {
```

Comment:
Just want to bring up that if lnd crashes before or during this step, the caller will need to subscribe to events again. I don't think this needs to be handled on the lnd end though - wdyt?

Reply:
No, the caller should just re-subscribe and then call `LookupHtlc` to see if they missed something for the htlcs that they are interested in. It needs to happen in that order though to prevent race conditions.

Reply:
I think it won't be easy. The htlc notifier was designed only for peeking into the current stream of events and not retrieve a full log of everything that happened. See for example https://github.com/lightningnetwork/lnd/pull/3781#issuecomment-588174538.